flutter/cpp/datasets/coco_gen.cc

```diff
@@ -0,0 +1,90 @@
+#include "coco_gen.h"
+
+#include <iomanip>
+
+namespace mlperf {
+namespace mobile {
+namespace {}  // namespace
+
+CocoGen::CocoGen(Backend* backend, const std::string& input_tfrecord,
+                 const std::string& input_clip_model)
+    : Dataset(backend),
+      sample_reader_(input_tfrecord),
+      clip_score_predictor_(input_clip_model),
+      samples_(sample_reader_.Size()) {}
+
+void CocoGen::LoadSamplesToRam(const std::vector<QuerySampleIndex>& samples) {
+  for (QuerySampleIndex sample_idx : samples) {
+    tensorflow::tstring record = sample_reader_.ReadRecord(sample_idx);
+    samples_.at(sample_idx) =
+        std::move(std::make_unique<CaptionRecord>(CaptionRecord(record)));
+  }
+}
+
+void CocoGen::UnloadSamplesFromRam(
+    const std::vector<QuerySampleIndex>& samples) {
+  for (QuerySampleIndex sample_idx : samples) {
+    samples_.at(sample_idx).release();
+  }
+}
+
+#define OUTPUT_SIZE 512 * 512 * 3
+std::vector<uint8_t> CocoGen::ProcessOutput(const int sample_idx,
+                                            const std::vector<void*>& outputs) {
+  void* output = outputs.at(0);
+  std::vector<uint8_t> output_pixels(OUTPUT_SIZE);
+  if (output_format_[0].type == DataType::Uint8) {
+    uint8_t* temp_data = reinterpret_cast<uint8_t*>(output);
+    std::copy(temp_data, temp_data + output_pixels.size(),
+              output_pixels.begin());
+  } else if (output_format_[0].type == DataType::Float32) {
+    float* temp_data = reinterpret_cast<float*>(output);
+
+    // [-1.0, 1.0] -> [0, 255]
+    for (int i = 0; i < OUTPUT_SIZE; i++) {
+      output_pixels[i] = (uint8_t)((*(temp_data + i) + 1) / 2 * 255);
+    }
+  }
+  if (!output_pixels.empty()) {
+    CaptionRecord* record = samples_.at(sample_idx).get();
+    auto input_ids = record->get_input_ids_vector();
+    auto attention_mask = record->get_attention_mask_vector();
+    // TODO: resize output_pixels to 224x224 and assign to pixel_values
```

Comment:
@freedomtan Do you have any suggestion how we can resize the image here?
The `tflite::evaluation::ImagePreprocessingStage` can only read from file path, so I cannot use it here.

Reply:
I resolve this one by adding a resizing layer to the CLIP model. The notebook to convert the CLIP model to TFLite is included in the PR.

Reply:
> my old resize() with TFLite resize op (https://github.com/tensorflow/tensorflow/blob/2ca3e974e350664c0e18b2ff5a52867490a15222/tensorflow/lite/examples/label_image/bitmap_helpers_impl.h#L30)

In code, we will need to have 2 steps: resize image and convert data layout  from NHWC to NCHW. This is already integrated in the TFLite model.
After some consideration, I think it's better to have the steps integrated in the TFLite model.